src/utils/db.ts

```diff
@@ -0,0 +1,48 @@
+import { Level } from "level";
+
+export type DBLevel = Level<string, string>;
+
+export default class DBService {
+  protected db: DBLevel;
+
+  private static _instance: DBService = new DBService();
+
+  protected constructor() {
+    if (DBService._instance) {
+      throw new Error(
+        "Error: Instantiation failed: Use DBService.getInstance() instead of new."
+      );
+    }
+    DBService._instance = this;
+
+    if (process.env.NODE_IS_TEST === "true") {
+      this.db = new Level<string, string>("./database_test", {
+        valueEncoding: "json",
+        createIfMissing: true,
+        errorIfExists: false,
```

Comment:
all this logic feels duplicated. The only difference i see between the 2 is the db name

why not doing:
```ts
const dbName = process.env.NODE_IS_TEST ? 'database_test' : 'database'
his.db = new Level<string, string>(dbName, {
        valueEncoding: "json",
        createIfMissing: true,
        errorIfExists: false,
      });
```

Reply:
Now respects paths passed via the constructor.